### PR TITLE
Add has_check? to Module & Scanner classes

### LINKED ISSUE #12660: Search returns incorrect aux module Check flag
## Steps to reproduce
1. `search type:auxiliary`
2. `use 1083` (This uses the SIP Invite Spoof module, which is displayed as supporting check by Search)
3. `check`
4. You should get a message saying that "This module does not support check."

## Expected behavior

`This module does not support check` should be displayed in step 4

## Current behavior

Search returns incorrect aux module Check flag

## System stuff

As seen in PR [12627](https://github.com/rapid7/metasploit-framework/pull/12627), `info` used to incorrectly report that `check` was supported for Auxiliary modules that used the `Msf::Auxiliary::Scanner` mixin. Odds are the fix used in 12627 will need to be applied to the logic in `search` that determines if a module supports `check` for the same reason.

### Metasploit version

Framework: 5.0.63-dev-104710ca41
Console  : 5.0.63-dev-104710ca41

### I installed Metasploit with:
- [ ] Source install (ruby version 2.6.5)

### OS
MacOS 10.14.6

### PR DESCRIPTION
**Pull request by agalway-r7**
Fixes #12660  

A lot of auxiliary modules incorrectly report that they support the check method due to their dependency on the 'Scanner' module and it also using a 'check' method.

These changes add the `has_check?` method to the `Module` class that supports checking if a module responds to `:check`. This method is then overwritten in `Scanner` to determine if the module responds to `:check_host`

This is a repeat of PR #12695, as it displayed the wrong status for modules that support `check` through mixins.

The large number of changes for `module.rb` is fixing indentation for the class, my changes are the added method in lines 146-148.

## Verification

List the steps needed to make sure this thing works

- [ ] checkout `master`
- [ ] `./msfconsole`
- [ ] `search type:auxiliary`
- [ ] Item 1082 (SIP Invite Spoof) supports check according to `search` 
- [ ] `use 1082`
- [ ] `check`
- [ ] `This module does not support check.` should be displayed
- [ ] `exit`
---
- [ ] checkout my branch
- [ ] `rm db/modules_metadata_base.json`
- [ ] `bundle install`
- [ ] `./msfconsole`
- [ ] `search type:auxiliary`
- [ ] Item 1082 (SIP Invite Spoof) no longer supports check according to `search`

### CODE REVIEW

**Review by wvu — APPROVED** [review]
I quite like this solution. This is more what it should be (and should have been post-merge of #10341).

**Comment by wvu-r7** [comment]
Some other cases that may benefit from calling `has_check?`:

```
wvu@kharak:/rapid7/metasploit-framework:HEAD$ git grep 'respond_to.*check'
lib/msf/base/serializer/readable_text.rb:    output << "#{indent}#{mod.respond_to?(:check) ? 'Yes' : 'No'}\n\n"
lib/msf/base/serializer/readable_text.rb:    output << "#{indent}#{mod.respond_to?(:check) ? 'Yes' : 'No'}\n\n"
lib/msf/base/serializer/readable_text.rb:      if session.respond_to?(:last_checkin) && session.last_checkin
lib/msf/base/serializer/readable_text.rb:      if session.respond_to?(:last_checkin) && session.last_checkin
lib/msf/base/simple/auxiliary.rb:            m.respond_to?(:check) ? m.check : Msf::Exploit::CheckCode::Unsupported
lib/msf/base/simple/auxiliary.rb:        m.respond_to?(:check) ? m.check : Msf::Exploit::CheckCode::Unsupported
lib/msf/base/simple/exploit.rb:      result = mod.respond_to?(:check) ? mod.check : Msf::Exploit::CheckCode::Unsupported
lib/msf/core/auxiliary/scanner.rb:  respond_to?(:check_host)
lib/msf/core/module.rb:      respond_to?(:check)
lib/msf/ui/console/command_dispatcher/modules.rb:                      o.respond_to?(:check) ? 'Yes' : 'No',
lib/msf/ui/console/module_command_dispatcher.rb:      if instance.respond_to?(:check_simple)
modules/exploits/linux/misc/quest_pmmasterd_bof.rb:    unless respond_to?(target[:check], true)
modules/post/multi/recon/local_exploit_suggester.rb:      mod.respond_to?(:check) &&
plugins/beholder.rb:      return unless framework.sessions[sid].respond_to?(:last_checkin)
spec/lib/msf/db_manager_spec.rb:    it { is_expected.to respond_to :check }
spec/lib/msf/ui/console/command_dispatcher/db_spec.rb:  it { is_expected.to respond_to :db_check_driver }
spec/lib/rex/payloads/meterpreter/uri_checksum_spec.rb:  it { is_expected.to respond_to :generate_uri_checksum}
spec/lib/rex/payloads/meterpreter/uri_checksum_spec.rb:  it { is_expected.to respond_to :uri_checksum_lookup}
spec/support/shared/examples/msf/module/module_info.rb:  it { is_expected.to respond_to_protected :merge_check_key }
wvu@kharak:/rapid7/metasploit-framework:HEAD$
```

**Comment by wvu-r7** [comment]
For reviewers: https://github.com/rapid7/metasploit-framework/pull/12702/files?w=1 to ignore the whitespace changes in the diff.

**Comment by adamgalway-r7** [comment]
Above commit replaces usages of `respond_to?` with `has_check?` where appropriate as suggested by @wvu-r7

**Review by wvu — APPROVED** [review]

File: lib/msf/core/module.rb [hunk 1]
```diff
@@ -13,402 +13,407 @@ module Msf
...
+        'Notes'       => {}
+      }.update(self.module_info)
+      self.module_store = {}
+    end
+
+    attr_writer   :platform, :references # :nodoc:
+    attr_writer   :privileged # :nodoc:
+    attr_writer   :license # :nodoc:
+
+  end
```
Comment (wvu-r7) [inline lib/msf/core/module.rb]: Hehe.

**Comment by wvu-r7** [comment]
# Release Notes

This creates the `has_check?` method for modules to determine if a module supports checking.

**Pull request merged**